### PR TITLE
feat: add release process via github actions

### PR DESCRIPTION
**Pull request by mrice32**
Just adds a simple config using github's recommended process whereby it publishes when you create a release on the github UI.

### CODE REVIEW

**Review by chrismaree — APPROVED** [review]
nice!

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      - uses: actions/checkout@v2
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16.x"
+          registry-url: "https://registry.npmjs.org"
+      - run: yarn
+      - run: yarn publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (chrismaree) [inline .github/workflows/publish.yml:18]: how is this set?

Comment (mrice32) [reply .github/workflows/publish.yml:18]: I set one in the across github org. It can be set at the org or repo level.

Comment (mrice32) [reply .github/workflows/publish.yml:18]: One thing I want to better understand is who has access to these secrets via PRs.

**Pull request merged**